gym/lib/gym/generators/build_command_generator.rb

```diff
@@ -39,7 +39,9 @@ def options
         options << "-destination '#{config[:destination]}'" if config[:destination]
         options << "-archivePath #{archive_path.shellescape}" unless config[:skip_archive]
         options << "-resultBundlePath '#{result_bundle_path}'" if config[:result_bundle]
-        options << "-scmProvider system" if config[:use_system_scm]
+        if config[:use_system_scm] && !options.include?("-scmProvider system")
```

Comment:
I think we should improve the entire flow to compose xcodebuild options to no have duplicated options but this is fine for now; e.g. Clarify what `project_path_array` brings and use Hash to manage options and convert them to real command argument eventually.

Thank you for taking care of this!